metrics-httpclient/src/test/java/io/dropwizard/metrics/httpclient/InstrumentedHttpClientsTest.java

```diff
@@ -41,4 +49,39 @@ public void registersExpectedMetricsGivenNameStrategy() throws Exception {
 
         verify(registryListener).onTimerAdded(eq(metricName), any(Timer.class));
     }
+
+    @Test
+    public void registersExpectedExceptionMetrics() throws Exception {
+        ServerSocket server = new ServerSocket();
+        server.bind(new InetSocketAddress("localhost", 0));
+        final HttpGet get = new HttpGet("http://localhost:" + server.getLocalPort() + "/");
+        final MetricName requestMetricName = MetricName.build("request");
+        final MetricName exceptionMetricName = MetricName.build("exception");
+
+        Thread serverThread = new Thread(() -> {
+            try {
+                final Socket socket = server.accept();
+                socket.close();
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+        });
+        serverThread.start();
+
+        when(metricNameStrategy.getNameFor(anyString(), any(HttpRequest.class)))
+            .thenReturn(requestMetricName);
+        when(metricNameStrategy.getNameFor(anyString(), any(Exception.class)))
+            .thenReturn(exceptionMetricName);
+
+        try {
+            client.execute(get);
+            fail();
+        } catch (NoHttpResponseException expected) {
+
+        }
+        serverThread.interrupt();
+        serverThread.join();
```

Comment:
I think we could move the cleanup code to the `finally block` in case we *actually* have a failed test case.